Tests/js/views/tabs/assets/ez-locationviewlocationstabview-tests.js

```diff
@@ -222,6 +223,153 @@ YUI.add('ez-locationviewlocationstabview-tests', function (Y) {
         },
     });
 
+    fireSwitchVisibilityEventTest = new Y.Test.Case({
+        name: "eZ LocationViewLocationsTabView fire switch visibility event test",
+        setUp: function () {
+            this.contentMock = new Mock();
+            this.locationMock1 = new Mock();
+            this.locationMock2 = new Mock();
+            this.locations = [this.locationMock1, this.locationMock2];
+            this.refreshCalled = false;
+
+            Mock.expect(this.contentMock, {
+                'method': 'get',
+                'args': ['resources'],
+                returns: '/main/location/id'
+            });
+
+            this._initLocationMock(this.locationMock1, '43');
+            this._initLocationMock(this.locationMock2, '42');
+
+            this.view = new Y.eZ.LocationViewLocationsTabView({
+                content: this.contentMock,
+                container: '.container',
+                locations: this.locations,
+            });
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+            delete this.view;
+        },
+
+        _initLocationMock: function (locationMock, locationId) {
+            Mock.expect(locationMock, {
+                'method': 'get',
+                args: ['id'],
+                returns: locationId
+            });
+
+            Mock.expect(locationMock, {
+                'method': 'toJSON',
+                returns: {}
+            });
+        },
+
+        "Should trigger switchVisibility when tapping on the hide/reveal button": function () {
+            var that = this,
+                switchVisibilityFired = false;
+
+            this.view.render();
+            this.view.set('active', true);
+
+            this.view.on('switchVisibility', function () {
+                switchVisibilityFired = true;
+            });
+
+            this.view.get('container').one('.ez-locations-hidden-button').simulateGesture('tap', function () {
+                that.resume(function () {
+                    Assert.isTrue(switchVisibilityFired, "switchVisibility should have been called");
+                });
+            });
+            this.wait();
+        },
+
+        "Should provide the location to the `switchVisibility` event when tapping on the hide/reveal button": function () {
+            var that = this,
+                location = {},
+                button;
+
+            this.view.render();
+            this.view.set('active', true);
+
+            this.view._refresh = function () {
```

Comment:
monkey patching is bad and not needed here. To check if the view is refreshed, you just have to check that the `loadLocations` event is fired.